### PR TITLE
Placeable pumpkin pies - Hallowed Edition

### PR DESCRIPTION
**Pull request by YanisBft**
Second attempt of #153

### CODE REVIEW

File: src/main/java/com/fabriccommunity/thehallow/block/PumpkinPieBlock.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+	public PumpkinPieBlock(Settings settings) {
+		super(settings);
+	}
+
+	public VoxelShape getOutlineShape(BlockState state, BlockView blockView, BlockPos pos, EntityContext entityContext) {
+		return SHAPE;
+	}
+
+	public BlockState getStateForNeighborUpdate(BlockState state_1, Direction direction, BlockState state_2, IWorld iWorld, BlockPos pos_1, BlockPos pos_2) {
```
Comment (vini2003) [inline src/main/java/com/fabriccommunity/thehallow/block/PumpkinPieBlock.java]: Those variable names are the only issue with this ATM.

**Review by Shnupbups — CHANGES_REQUESTED** [review]
I'd like the Pumpkin Pie block to be edible like Cake, with a bonus for if the Pumpkin Ring is worn.

**Review by liach — APPROVED** [review]

**Comment by YanisBft** [comment]
So now, pumpkin pie block can be eaten by quarters. I did what I could with the models, sorry about that ^^'

Also, note that unlike cakes, `bites` blockstate represents the remaining bites instead of the eaten bites, which makes more sense to me.

File: src/main/java/com/fabriccommunity/thehallow/registry/HallowedBlocks.java [hunk 2]
```diff
@@ -118,6 +118,8 @@
 	public static final Block DECEASED_MOSS = register("deceased_moss", new Block(FabricBlockSettings.copy(Blocks.PODZOL).materialColor(MaterialColor.PURPLE).build()));
 	public static final Block HALLOWED_GATE = register("hallowed_gate", new HallowedGateBlock(FabricBlockSettings.copy(Blocks.ENCHANTING_TABLE).materialColor(MaterialColor.BROWN).build()));
 	
+	public static final Block PUMPKIN_PIE = Registry.register(Registry.BLOCK, TheHallow.id("pumpkin_pie"), new PumpkinPieBlock(FabricBlockSettings.of(Material.CAKE).strength(0.5F, 0.5F).sounds(BlockSoundGroup.WOOL).build()));
```
Comment (Shnupbups) [inline src/main/java/com/fabriccommunity/thehallow/registry/HallowedBlocks.java]: Please use the registry method that already exists in HallowedBlocks, like other blocks use.

Comment (YanisBft) [reply src/main/java/com/fabriccommunity/thehallow/registry/HallowedBlocks.java]: Done

File: src/main/java/com/fabriccommunity/thehallow/block/PumpkinPieBlock.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+	}
+	
+	public VoxelShape getOutlineShape(BlockState state, BlockView blockView, BlockPos pos, EntityContext entityContext) {
+		return SHAPE[state.get(BITES) - 1];
+	}
+	
+	public boolean activate(BlockState state, World world, BlockPos pos, PlayerEntity player, Hand hand, BlockHitResult hitResult) {
+		if (!world.isClient) {
+			return this.tryEat(world, pos, state, player);
+		} else {
```
Comment (liach) [inline src/main/java/com/fabriccommunity/thehallow/block/PumpkinPieBlock.java]: consider dropping `else` after `return` to keep code clean

File: src/main/java/com/fabriccommunity/thehallow/block/PumpkinPieBlock.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+			ItemStack stack = player.getStackInHand(hand);
+			return this.tryEat(world, pos, state, player) || stack.isEmpty();
+		}
+	}
+	
+	private boolean tryEat(IWorld iWorld, BlockPos pos, BlockState state, PlayerEntity player) {
+		if (!player.canConsume(false)) {
+			return false;
+		} else {
```
Comment (liach) [inline src/main/java/com/fabriccommunity/thehallow/block/PumpkinPieBlock.java:68]: Same, drop `else` after `return`

**Review by liach — APPROVED** [review]

**Review by Shnupbups — APPROVED** [review]

**Review by vini2003 — APPROVED** [review]
Same issues as before. Just needs IDEA reformatter run over it some day.

**Pull request merged**